src/Symfony/Contracts/Deprecation/deprecated.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * Triggers a deprecation.
+ *
+ * As recommended for prod, turn the "zend.assertions" ini setting to 0 or -1 to disable deprecation notices.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+function deprecated($package, $version, $message)
```

Comment:
The function doesn't use type hints nor variadics on purpose, to make it as fast as possibly achievable.

Reply:
I'm not sure it makes a real difference. 
I already benched this, and the difference is small: https://github.com/lyrixx/php-bench/commit/ea0ff97ad01b91e554377760028f4588beba25d7
And It's not gonna be hit that much.

Reply:
I've already seen the same deprecation triggered in a loop million times.
And there is no benefit to using types here.